### PR TITLE
Implement proper renameFile on Windows

### PR DESCRIPTION
**Pull request by gabriel-samfira**
Symlinks on Windows are implemented using NTFS reparse points. Reparse points are also used when creating volume mounts on Windows. As far as golang is concerned, mountpoints and symlinks look the same. This change makes the distinction between the two and allows safely renaming symlinks, while erring out for anything else that looks like a symlink.

Signed-off-by: Gabriel Adrian Samfira <gsamfira@cloudbasesolutions.com>

### CODE REVIEW

File: diskwriter_windows.go [hunk 1]
```diff
@@ -1,8 +1,15 @@
+//go:build windows
 // +build windows
 
 package fsutil
 
 import (
+	"fmt"
+	ioFS "io/fs"
```
Comment (tonistiigi) [inline diskwriter_windows.go:8]: what's up with this package alias. `fs` doesn't even seem to be used elsewhere in the file.

Comment (gabriel-samfira) [reply diskwriter_windows.go:8]: remnant from splitting up the previous PR. Changing it now.

Comment (gabriel-samfira) [reply diskwriter_windows.go:8]: The only way we can import ```io/fs``` is if we rename that struct. I think the aliased import is probably the path with the least amount of code changes :smile: . What do you think?

Comment (tonistiigi) [reply diskwriter_windows.go:8]: Ok, but make it all lowercase.

Comment (gabriel-samfira) [reply diskwriter_windows.go]: Import was renamed. Please take another look.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**